### ISSUE TITLE
Access the character used in task list

### ISSUE THREAD

**Issue by felipesere** [kivikakk/comrak#251]
I want to use comrak to parse some markdown that I currently write in ObsidianMD.
In Obsidian, I can have different visuals on tasks depending on the character used.
E.g.
```markdown
* [ ] looks unchecked
* [x] is checked
* [!] is still checked, but with a ⚠️
* [/] looks "partially done"
```

if would be nice if the character itself  was exposed so I can properly roundtrip the markdown after some processing.

**Comment by kivikakk** [comment]
I'm very open to this idea! Pull requests gratefully accepted.

**Comment by felipesere** [comment]
I think it comes down to capturing the character here https://github.com/kivikakk/comrak/blob/main/src/parser/mod.rs#L1653 and tracking in the struct?
I'll get to it this weekend 😅

**Comment by kivikakk** [comment]
Yep, that looks exactly right. 👍 😊

**Comment by felipesere** [comment]
👋 First attempt :)

https://github.com/kivikakk/comrak/pull/252

**Comment by digitalmoksha** [comment]
@felipesere just as an fyi, GitLab implemented the `[~]` syntax, which indicates that a task item is inapplicable, and would typically be displayed as being struck out.

https://docs.gitlab.com/ee/user/markdown.html#task-lists

**Comment by felipesere** [comment]
Fixed in 536471b86f921e3c136501edabedfcb6df6c4715
